src/bindgen/parser/method.cr

```diff
@@ -416,45 +421,191 @@ module Bindgen
+          const: @const,
+          virtual: @virtual,
+          pure: @pure,
+          extern_c: @extern_c,
+          class_name: @class_name,
+          arguments: [] of Argument,
+          return_type: @return_type,
+          crystal_name: crystal_name,
+        )
+        fixed.binding_name = binding_method_name
+        fixed
+      end
+
       # Mangled name for the C++ wrapper method name
       def mangled_name
         class_name = Util.mangle_type_name(@class_name)
-        "bg_#{class_name}_#{binding_method_name}_#{binding_arguments_name}"
+        method_name = Util.mangle_type_name(binding_method_name)
+        args_name = binding_arguments_name
+
+        "bg_#{class_name}_#{method_name}_#{args_name}"
       end
 
       # Mangles the list of argument types into a flat string.
-      def binding_arguments_name
+      private def binding_arguments_name
         @arguments.map(&.mangled_name).join("_")
       end
 
-      # Name of the method in C++ and Crystal bindings.
-      def binding_method_name
-        name = Util.mangle_type_name(@name)
+      # Name of the method in C++ and Crystal bindings.  If an explicit
+      # `#binding_name` name is set, it'll be returned without further
+      # processing.
+      private def binding_method_name
+        if enforced_name = @binding_name
+          return enforced_name
+        end
 
         case self
-        when .constructor?           then "#{name}_CONSTRUCT"
-        when .aggregate_constructor? then "#{name}_AGGREGATE"
-        when .copy_constructor?      then "#{name}_COPY"
-        when .member_getter?         then "#{name}_GETTER"
-        when .member_setter?         then "#{name}_SETTER"
-        when .operator?              then operator_name
-        when .static_method?         then "#{name}_STATIC"
-        when .static_getter?         then "#{name}_STATIC_GETTER"
-        when .static_setter?         then "#{name}_STATIC_SETTER"
-        when .destructor?            then "#{name}_DESTROY"
-        else                              name
+        when .constructor?           then "_CONSTRUCT"
+        when .aggregate_constructor? then "_AGGREGATE"
+        when .copy_constructor?      then "_COPY"
+        when .member_getter?         then "#{@name}_GETTER"
+        when .member_setter?         then "#{@name}_SETTER"
+        when .operator?              then "_OPERATOR_#{binding_operator_name}"
+        when .static_method?         then "#{@name}_STATIC"
+        when .static_getter?         then "#{@name}_STATIC_GETTER"
+        when .static_setter?         then "#{@name}_STATIC_SETTER"
+        when .destructor?            then "_DESTROY"
+        else                              @name
+        end
+      end
+
+      # Converts *name* to an operator method in Crystal wrappers.
+      private def to_crystal_operator_name(name)
+        return "call" if name == "operator()"
+
+        case @arguments.size
+        when 0 then to_crystal_operator1_name(name)
```

Comment:
I think instead of 1/2 the more formal `unary` and `binary` would be better. E.g. `#to_unary_crystal_operator_name`